lib/task_manager.js

```diff
@@ -0,0 +1,84 @@
+class TaskManager {
+  constructor() {
```

Comment:
could you make it private?

Reply:
To make it private, I'd need to migrate this to Typescript, but I'd do that in a following PR. JS doesn't allow private classes